### PR TITLE
Add option to install the tracker as an imported target in a CMake project (close #85)

### LINKED ISSUE #85: Want to link static library with my application/lib
**Is your feature request related to a problem? Please describe.**
I want to link snowplow static library with my application code but it seems configuration is fixed by CMakeFiles.txt

I am using FetchContent method.

**Describe the solution you'd like**
I'd like to link with snowplow static lib by using an easy way.

**Describe alternatives you've considered**
I've tryed to link by using snowplow.lib but includes break.  

I also tried to add include paths to headers and libs but it misses nlohmann dependencies.

**Additional context**
I want to have the smaller binaries as possible.

### PR DESCRIPTION
**Pull request by tamaskenezlego**
This PR extends the project's CMakeLists.txt to install the Snowplow headers, library and a cmake config file to allow client projects to import Snowplow with `find_package(snowplow)`.

It's only available if

- `nlohmann_json` is an external dependency (when `BUILD_SHARED_LIBS=ON`), or
- also `sqlite3` is an external dependencies (when `BUILD_SHARED_LIBS=OFF`)

Besides adding a cmake install target I made these changes:

- CMake minimum version raised from 3.14 to 3.15 to make use of the `CMAKE_FIND_PACKAGE_PREFER_CONFIG` feature which allows for example `find_package(CURL)` to find a Conan-installed libcurl even if the system also provides one.
- Changed SQlite3 and libcurl and libUUID to a private dependency since they're not really used in the headers.
- Since on Macos Snowplow doesn't use libcurl I changed the relevant conditions to omit it on Macos. I still kept libcurl as a Macos dependency for `snowplow-tests`.
- Updated README.md
- Renamed and updated `FindUUID.cmake` to `FindLibUUID.cmake` to conform to the `libuuid-config.cmake` provided by Conan.

The test project makes the dependencies available using Conan and builds a minimal program which links to Snowplow.

The workflow is a matrix job with these dimensions:

- OS: mac/win/linux
- build shared libs: yes/no

For Linux, the job also tests the configurations when libcurl and libuuid is used as a system dependency or as a conan package.

### CODE REVIEW

File: .github/workflows/install.yml
Comment (matus-tomlein) [inline .github/workflows/install.yml]: Is there any specific reason why you didn't make use of Conan version 2? I understand this is only internal usage, but would be great to have the dependency up to date if possible.

File: include/snowplow/http/http_client_curl.hpp
Comment (matus-tomlein) [inline include/snowplow/http/http_client_curl.hpp:16]: I agree that we don't need to provide the CURL client on macOS as it just complicates the build, but it is a breaking change from the previous version so we'll need to change the tracker version to v2. I was wondering if you know of any other breaking changes in this PR that we should mention in the upgrading guide to the next version?

File: test/install/main.cpp
Comment (matus-tomlein) [inline test/install/main.cpp]: Just wondering if there is any way we could make use of the example app in `examples/main.cpp` for this purpose? It would just be easier to maintain in the future to only have one such app although I understand that the two apps serve different purposes (one is to showcase the tracking APIs and one to test the build).

File: .github/workflows/install.yml
Comment (matus-tomlein) [inline .github/workflows/install.yml:1]: Love this new workflow!

**Review by matus-tomlein — COMMENTED** [review]
This is amazing work, @tamaskenezlego! A huge leap forward for the tracker. I just had a few questions before merging this in.

File: .github/workflows/install.yml
Comment (tamaskenezlego) [reply .github/workflows/install.yml]: I had this experience that many recipes didn't work with Conan 2. But that was right after the release. I'll try to upgrade to V2, the depencies here are quite common, should work with V2.

File: include/snowplow/http/http_client_curl.hpp
Comment (tamaskenezlego) [reply include/snowplow/http/http_client_curl.hpp:16]: Let me know if we should put back CURL so you need to raise a major version. I'll think this over if there are any other changes.

Comment (tamaskenezlego) [reply include/snowplow/http/http_client_curl.hpp:16]: EDIT: Let me know if we should put back CURL so you **_don't_** need to raise a major version.

Other possible breaking changes:

- CMake minimum version changed 3.14 -> 3.15
- As you mentioned libcurl has been removed as a dependency of Snowplow on Macos. If a client project (using Snowplow as a subdirectory) relies on getting libcurl through Snowplow then it will fail now.
- A bit similar: sqlite3, libcurl, libuuid are now PRIVATE dependencies. If a client project relies on getting the sqlite3/libcurl/libuuid include paths through Snowplow then it will fail now.

File: .github/workflows/install.yml
Comment (tamaskenezlego) [reply .github/workflows/install.yml]: It works with minimal changes: 30ffffeb0373994a8149ef8e0269578790cc3d36

Comment (matus-tomlein) [reply .github/workflows/install.yml]: That's great!

File: include/snowplow/http/http_client_curl.hpp
Comment (matus-tomlein) [reply include/snowplow/http/http_client_curl.hpp:16]: Thanks for listing the other breaking changes! I think it's fine to raise the major version in this case.

File: test/install/main.cpp
Comment (matus-tomlein) [reply test/install/main.cpp]: @tamaskenezlego Happy to merge this regardless of this comment unless you have something else in mind?

Comment (tamaskenezlego) [reply test/install/main.cpp]: I'm just trying out how this would work, I'll get back to you soon.

Comment (tamaskenezlego) [reply test/install/main.cpp]: Ok, I think it was a good idea, I was able to do it without any compromises. Now `examples/main.cpp` is built both as part of the main project and in standalone mode, in the install test workflow. 679f955a16f49eba563067bf525c57c04cd3fa76
Please review and if it's fine you can merge.

Comment (matus-tomlein) [reply test/install/main.cpp]: This is perfect, so much cleaner! Just a minor thing – could you update the README? It says to run the example app using `./snowplow-example {{ your collector uri }}` from inside the build folder, but now I think it is `./examples/snowplow-example {{ your collector uri }}`

**Review by matus-tomlein — APPROVED** [review]
LGTM! 👍 💯

File: test/install/main.cpp
Comment (tamaskenezlego) [reply test/install/main.cpp]: You're right, fixed it! a1439d3a0f7b2e48dfa1cb879ba0d5773aaa0b0f
Also, I saw there's a Windows section which seems to be outdated since there are no `.sln` files in the repo anymore. Not sure what are your plans with that.

Comment (matus-tomlein) [reply test/install/main.cpp]: Thanks! I'll make a note to update the Windows instructions in a separate PR.

**Pull request merged**
